as/cNFT/assembly/core.ts

```diff
@@ -35,16 +36,42 @@ export function nft_transfer(token_id: TokenId, bidder_id: AccountId): void {
     persistent_tokens.add(token_id, token, bidder_id)
 
     /* Deleting token from previous owner */
-    persistent_tokens.remove(token.id, token.prev_owner_id)
+    persistent_tokens.remove_from_account(token.id, token.prev_owner_id)
+
 
-    
     // Immiting log event
     const transfer_log = new NftTransferLog()
-    
+
     transfer_log.old_owner_id = token.prev_owner_id
     transfer_log.new_owner_id = token.owner_id
     transfer_log.token_ids = [token.id]
 
     const log = new NftEventLogData<NftTransferLog>('nft_transfer', [transfer_log])
     logging.log(log)
+
+}
+
+
+@nearBindgen
+export function burn_design(token_id: TokenId): void {
+
+    /* Getting stored token from tokenId */
+    const token = persistent_tokens.get(token_id)
+
+    assert(context.sender == token.owner_id, "You must be the owner of the token to burn")
+
+    /* Deleting token, metadata and royalty */
+    persistent_tokens.remove(token_id, token.owner_id);
+    persistent_tokens_metadata.remove(token_id);
+    persistent_tokens_royalty.remove(token_id);
+
+    // Immiting log event
+    const burn_log = new NftBurnLog();
+
+    burn_log.owner_id = token.owner_id;
+    burn_log.authorized_id = context.sender;
+    burn_log.token_ids = [token.id];
+
+    const log = new NftEventLogData<NftBurnLog>('nft_burn', [burn_log]);
+    logging.log(log);
```

Comment:
Would also to also burn market:
https://github.com/curaOS/contracts/blob/dec82c12a54ed487e3dab8e62cd7c28ba15d933b/as/Market/assembly/index.ts#L192

Reply:
Not relevant